dspace-api/src/main/java/org/dspace/app/orcid/consumer/OrcidQueueConsumer.java

```diff
@@ -0,0 +1,354 @@
+import org.dspace.core.Context;
+import org.dspace.event.Consumer;
+import org.dspace.event.Event;
+import org.dspace.services.ConfigurationService;
+import org.dspace.services.factory.DSpaceServicesFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The consumer to fill the ORCID queue. The addition to the queue is made for
+ * all archived items that meet one of these conditions:
+ * <ul>
+ * <li>are profiles already linked to orcid that have some modified sections to
+ * be synchronized (based on the preferences set by the user)</li>
+ * <li>are publications/fundings related to profile items linked to orcid (based
+ * on the preferences set by the user)</li>
+ * 
+ * </ul>
+ *
+ * @author Luca Giamminonni (luca.giamminonni at 4science.it)
+ *
+ */
+public class OrcidQueueConsumer implements Consumer {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(OrcidQueueConsumer.class);
+
+    private OrcidQueueService orcidQueueService;
+
+    private OrcidHistoryService orcidHistoryService;
+
+    private OrcidTokenService orcidTokenService;
+
+    private OrcidSynchronizationService orcidSynchronizationService;
+
+    private ItemService itemService;
+
+    private OrcidProfileSectionFactoryService profileSectionFactoryService;
+
+    private ConfigurationService configurationService;
+
+    private RelationshipService relationshipService;
+
+    private List<UUID> alreadyConsumedItems = new ArrayList<>();
+
+    @Override
+    public void initialize() throws Exception {
+
+        OrcidServiceFactory orcidServiceFactory = OrcidServiceFactory.getInstance();
+
+        this.orcidQueueService = orcidServiceFactory.getOrcidQueueService();
+        this.orcidHistoryService = orcidServiceFactory.getOrcidHistoryService();
+        this.orcidSynchronizationService = orcidServiceFactory.getOrcidSynchronizationService();
+        this.orcidTokenService = orcidServiceFactory.getOrcidTokenService();
+        this.profileSectionFactoryService = orcidServiceFactory.getOrcidProfileSectionFactoryService();
+        this.configurationService = DSpaceServicesFactory.getInstance().getConfigurationService();
+        this.relationshipService = ContentServiceFactory.getInstance().getRelationshipService();
+
+        this.itemService = ContentServiceFactory.getInstance().getItemService();
+    }
+
+    @Override
+    public void consume(Context context, Event event) throws Exception {
+        DSpaceObject dso = event.getSubject(context);
+        if (!(dso instanceof Item)) {
+            return;
+        }
+        Item item = (Item) dso;
+        if (!item.isArchived()) {
+            return;
+        }
+
+        if (alreadyConsumedItems.contains(item.getID())) {
+            return;
+        }
+
+        if (isOrcidSynchronizationDisabled()) {
+            return;
+        }
+
+        context.turnOffAuthorisationSystem();
+        try {
+            consumeItem(context, item);
```

Comment:
@LucaGiamminonni and me discussed this on slack. If the item is the item referenced by the event object, uncaching it would affect following consumers. If the item is initialized by `findAllRelatedItems(...)` following consumers would not be affected. Tracking this is probably to much to ask, unless we see real performance issues, which are rather unlikely here.